### PR TITLE
Fix/1520 next five subtests

### PR DESCRIPTION
**Pull request by spazm**
### Summary
Convert next 5 tests to use subtests:
methods.t, reactor_ev.t, user_agent_socks.t, websocket.t, and websocket_frames.t 

### Motivation
Requests in #1520 

### References
Requested in #1520 

There are a couple of blocks of subtests that don't seem to have tests.  They may have just been labeled parts of setting up?  Please feel free to revert/modify as necessary.

### CODE REVIEW

File: t/mojo/reactor_ev.t [hunk 1]
```diff
@@ -8,307 +8,356 @@ plan skip_all => 'EV 4.32+ required for this test!'                  unless eval
...
+  ok $triggered, 'reactor waited for one event';
+
+  my $time = time;
+  Mojo::IOLoop->start;
+  Mojo::IOLoop->one_tick;
+  ok time < ($time + 10), 'stopped automatically';
+};
+
+my ($listen,   $port)     = ();
+my ($readable, $writable) = ();
```
Comment (kraih) [inline t/mojo/reactor_ev.t:41]: These are outside the subtest.

File: t/mojo/websocket_frames.t [hunk 2]
```diff
@@ -4,274 +4,322 @@ use Test::More;
...
-$fragmented_compressed->on(message => sub { $text = pop });
-my $compressed_payload = $fragmented_compressed->build_message({text => 'just works'})->[5];
-ok !$text, 'message event has not been emitted yet';
-$fragmented_compressed->parse_message([0, 1, 0, 0, WS_TEXT, substr($compressed_payload, 0, 3)]);
-ok !$text, 'message event has not been emitted yet';
-$fragmented_compressed->parse_message([0, 0, 0, 0, WS_CONTINUATION, substr($compressed_payload, 3, 3)]);
-ok !$text, 'message event has not been emitted yet';
-$fragmented_compressed->parse_message([1, 0, 0, 0, WS_CONTINUATION, substr($compressed_payload, 6)]);
-is $text, 'just works', 'decoded correctly';
+my $dummy;
```
Comment (kraih) [inline t/mojo/websocket_frames.t:7]: Variable outside of subtest.

File: t/mojo/reactor_ev.t [hunk 3]
```diff
@@ -8,307 +8,356 @@ plan skip_all => 'EV 4.32+ required for this test!'                  unless eval
...
-my $port   = $listen->sockport;
-my ($readable, $writable);
-$reactor->io($listen => sub { pop() ? $writable++ : $readable++ })->watch($listen, 0, 0)->watch($listen, 1, 1);
-$reactor->timer(0.025 => sub { shift->stop });
-$reactor->start;
-ok !$readable, 'handle is not readable';
-ok !$writable, 'handle is not writable';
-
-# Connect
+my $reactor;
```
Comment (kraih) [inline t/mojo/reactor_ev.t:13]: Variable outside of subtest.

**Review by kraih — CHANGES_REQUESTED** [review]
Some tests are fine and could be merged, but others have scoping issues and require more attention.

**Comment by kraih** [comment]
I'm sorry, but i've decided to close these PRs, because they contain too many converted tests at once. And we don't have the review capacity to re-review the whole thing after every scoping issue fix.

**Comment by spazm** [comment]
Removed the tests that did not convert cleanly.  The remaining tests converted cleanly and required only minimal non-whitespace changes to redeclare variables for the new scopes.  PR description updated with exact details.

**Pull request closed without merging**